File: javascript/Mod2/2.07/test/LoginTest.js

// filename: test/LoginTest.js
const { Builder } = require('selenium-webdriver') 
const path = require('path')
const assert = require('assert')

//describe is a method from Mocha
describe('Login', function() { 
    this.timeout(30000)
    let driver

    beforeEach(async function() { 
        /* const vendorDirectory =
        path.delimiter + path.join(__dirname, '..', 'vendor') 
        process.env.PATH += vendorDirectory */
    driver = await new Builder().forBrowser('firefox').build()
    })
    afterEach(async function() { 
        await driver.quit()
    })
    it('with valid credentials', async function() {
        await driver.get('http://the-internet.herokuapp.com/login') 
        await driver
            .findElement({ id: 'username' })
            .sendKeys('tomsmith') 
        await driver
            .findElement({ id: 'password' })
            .sendKeys('SuperSecretPassword!')
        await driver.findElement({ css: 'button' }).click()
        assert(
            await driver.findElement({ css: '.flash.success' }).isDisplayed(), 'Success message not displayed'
        )
    }) 
})

File: python/Mod2/2.07/tests/login_test.py
# filename: tests/login_test.py
import pytest
import os
from selenium import webdriver
from selenium.webdriver.common.by import By

@pytest.fixture
def driver(request):
    _chromedriver= os.path.join(os.getcwd(), 'vendor', 'chromedriver')
    if os.path.isfile(_chromedriver):
        driver_ = webdriver.Chrome(_chromedriver)

    else:
        driver_ = webdriver.Chrome()

    def quit():
        driver_.quit()

    request.addfinalizer(quit)
    return driver_


def test_valid_credentials(driver):
    driver.get("http://the-internet.herokuapp.com/login")
    driver.find_element(By.ID, "username").send_keys("tomsmith")
    driver.find_element(By.ID, "password").send_keys("SuperSecretPassword!")
    driver.find_element(By.CSS_SELECTOR, "button").click()
    assert driver.find_element(By.CSS_SELECTOR, ".flash.success").is_displayed()